controllers/storagecluster/reconcile.go

```diff
@@ -651,9 +653,41 @@ func (r *StorageClusterReconciler) reconcilePhases(
 
 	}
 
+	// create a `ocs-metrics-exporter-ceph-auth` secret for metrics exporter
+	// create a cephclient and it will create the secret
+	err := r.createMetricsExporterCephClient(instance)
+	if err != nil {
+		r.Log.Error(err, "Failed to create ceph client for metrics exporter.")
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) createMetricsExporterCephClient(instance *ocsv1.StorageCluster) error {
+	cephClient := &rookCephv1.CephClient{}
+	cephClient.Name = "ocs-metrics-exporter-ceph-auth"
+	cephClient.Namespace = instance.Namespace
+
+	if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, cephClient, func() error {
+		if err := controllerutil.SetControllerReference(instance, cephClient, r.Scheme); err != nil {
+			return err
+		}
+		cephClient.Spec.SecretName = cephClient.Name
+		cephClient.Spec.Caps = map[string]string{
+			"mon": "profile rbd, allow command 'osd blocklist'",
+			"mgr": "allow rw",
+			"osd": "profile rbd",
+			"mds": "allow *",
```

Comment:
same here why do we need mds? i dont think we do something with cephfs pvc in metrics exporter.